internal/build/build.go

```diff
@@ -419,6 +424,23 @@ func linkMainPkg(ctx *context, pkg *packages.Package, pkgs []*aPackage, llFiles
 	err := ctx.env.Clang().Exec(args...)
 	check(err)
 
+	if runtime.GOOS == "darwin" {
+		dylibPaths := make([]string, 0, len(libs))
+		rpaths := make([]string, 0, len(libs))
+		for _, lib := range libs {
+			dylibPath := findDylibPath(app, lib)
+			if dylibPath != "" {
+				dylibPaths = append(dylibPaths, dylibPath)
+				rpaths = append(rpaths, filepath.Dir(dylibPath))
+			}
+		}
+		installNameToolCmd := install_name_tool.New("")
+		err := installNameToolCmd.ChangeToRpath(app, dylibPaths...)
+		check(err)
+		err = installNameToolCmd.AddRpath(app, rpaths...)
```

Comment:
```
          cmd LC_RPATH
      cmdsize 56
         path /opt/homebrew/Cellar/bdw-gc/8.2.6/lib (offset 12)
--
          cmd LC_RPATH
      cmdsize 48
         path /opt/homebrew/opt/bdw-gc/lib (offset 12)
```

This step adds rpaths similar to `/opt/homebrew/opt/bdw-gc/lib` (the actual dylib ID), which is distinct from those derived from `-L` (e.g., the `/opt/homebrew/Cellar/bdw-gc/8.2.6/lib`).

Reply:
why we need so may rpath(s)?

Reply:
This step is added because I believe it is more meaningful than the rpaths derived from `-L`, after all, they are dylib IDs and are usually not limited to specific versions.

But yes, that's too many rpaths (and will be more if dylibs are added). I'll remove this step.